File: folkmq/src/main/java/org/noear/folkmq/FolkMQ.java
package org.noear.folkmq;

import org.noear.folkmq.client.MqClient;
import org.noear.folkmq.client.MqClientDefault;
import org.noear.folkmq.server.MqServer;
import org.noear.folkmq.server.MqServerDefault;

/**
 * @author noear
 * @since 1.0
 */
public class FolkMQ {
    /**
     * 获取版本
     */
    public static String version() {
        return "1.0.20";
    }

    /**
     * 创建服务端
     */
    public static MqServer createServer() {
        return new MqServerDefault();
    }

    /**
     * 创建客户端
     */
    public static MqClient createClient(String... serverUrls) {
        return new MqClientDefault(serverUrls);
    }
}

File: folkmq-lang-js/src/FolkMQ.ts
import {MqClient, MqClientDefault} from "./client/MqClient";
import {IMqMessage, MqMessage} from "./client/IMqMessage";

export class FolkMQ {
    /**
     * 获取版本
     */
    static version(): string {
        return "1.2.0";
    }

    /**
     * 创建客户端
     */
    static createClient(serverUrls: string[]): MqClient {
        return new MqClientDefault(serverUrls);
    }

    static newMessage(content: string): IMqMessage {
        return new MqMessage(content);
    }
}